### PR TITLE
Create BaseTheme and add rounded buttons

### PR DESCRIPTION
**Pull request by biocom**
- Shared BaseTheme (colors, themeVariables)
- Rounded buttons for GP theme.

### CODE REVIEW

File: src/custom/theme/cowSwapTheme.tsx [hunk 1]
```diff
@@ -23,6 +24,7 @@ export * from '@src/theme/components'
 export function colors(darkMode: boolean): Colors {
   return {
     ...colorsUniswap(darkMode),
```
Comment (anxolin) [inline src/custom/theme/cowSwapTheme.tsx:26]: why do we need to still inherit from uniswap?

Comment (anxolin) [reply src/custom/theme/cowSwapTheme.tsx:26]: base colors inherit from uniswap already, isn't it?

**Comment by anxolin** [comment]
Also, @michelbio try to give names to the PRs. Instead of "Cow 3" give a name that would explain the scope of it

**Comment by biocom** [comment]
> Also, @michelbio try to give names to the PRs. Instead of "Cow 3" give a name that would explain the scope of it
> 
> On top of making reviews easier, the title of the PR goes later squashed into a commit, so it's nicer to be able to read the commit history and see what features are being implemented.

Will do!

File: src/custom/theme/cowSwapTheme.tsx (hunk 1 above)
Comment (biocom) [reply src/custom/theme/cowSwapTheme.tsx]: That's right, just fixed it !

**Review by anxolin — APPROVED** [review]
Nice!

For me now the question would be. How do I know if one style goes in the base or in GP. Is GP the base?

**Comment by biocom** [comment]
> Nice!
> 
> For me now the question would be. How do I know if one style goes in the base or in GP. Is GP the base?

For now my approach has been:
- Put properties/values which are shared between the current two themes, in the base theme, to keep the theme code dryer.
- Then any differences between the themes *and* the baseTheme should go in the individual theme file.

Let me know your thoughts.

**Pull request merged**